src/main/java/pl/betoncraft/betonquest/compatibility/citizens/CitizensParticle.java

```diff
@@ -75,22 +82,81 @@ public void run() {
 		enabled = true;
 	}
 
+
 	@Override
 	public void run() {
-		for (NPC npc : npcs) {
-			if (npc == null) {
-				continue;
-			}
-			Entity e = npc.getEntity();
-			if (e == null) {
-				continue;
-			}
-			Location loc = e.getLocation().clone();
-			loc.setPitch(-90);
-			Compatibility.getEffectManager().start(name, section, loc);
-		}
+	    // Bukkit.getLogger().info("Particle: RUN:89: Online Players: " + Bukkit.getOnlinePlayers().size() + "NPC:" + npcs.size());
+        for (NPC npc : npcs) {
+            if (npc == null) {
+                continue;
+            }
+            Entity e = npc.getEntity();
+            if (e == null) {
+                continue;
+            }
+            // Bukkit.getLogger().info("NPC Name: " + npc.getName());
+            Location loc = e.getLocation().clone();
+            loc.setPitch(-90);
+            // Compatibility.getEffectManager().start(name, section, loc);
+        }
+
+
+        Config.getPackages().forEach((n, c) -> activateEffects(n, c));
 	}
 
+    void activateEffects(String name, ConfigPackage cfgPack){
+        ConfigurationSection cfg = cfgPack.getCustom().getConfig().getConfigurationSection("particles");
+        if(cfg == null) return;
+
+
+        Set<String> keys = cfg.getKeys(false);
+
+        // Bukkit.getLogger().info("Keys : " + keys.size());
+        for(String key: keys){
+            // Bukkit.getLogger().info("Key: " + key);
+            String npcID = cfg.getString(key + ".NPC");
+            if(npcID == null){
+            	Bukkit.getLogger().warning("Malformed particles for NPC Key: " + key);
+                continue;
+            }
+
+            // Bukkit.getLogger().info("Particle: npcID:" + npcID + " Integer: " + Integer.parseInt(npcID));
+            NPC npc = CitizensAPI.getNPCRegistry().getById(Integer.parseInt(npcID));
+            if(npc == null) continue;
+            // Bukkit.getLogger().info("npcID: " + npcID + " toString(): " + Integer.toString(npc.getId()));
+
+            Location loc = npc.getStoredLocation().clone();
```

Comment:
Getting stored location will return a location even if the NPC is dead or isn't in a loaded chunk. It's not a good thing to display an effect in such circumstances. Getting the entity, checking if it's `null` and then getting the location is a lot safer.